ext/fetch/lib.rs

```diff
@@ -69,9 +74,24 @@ use tokio::io::AsyncWriteExt;
 // Re-export reqwest and data_url
 pub use data_url;
 pub use reqwest;
+pub use reqwest_middleware;
 
 pub use fs_fetch_handler::FsFetchHandler;
 
+static HTTP_MIDDLEWARE: Lazy<
+  Mutex<Vec<Arc<dyn reqwest_middleware::Middleware>>>,
+> = Lazy::new(|| Mutex::new(vec![]));
+
+pub fn add_middleware(
+  middleware: Arc<dyn reqwest_middleware::Middleware>,
+) -> Result<(), AnyError> {
+  HTTP_MIDDLEWARE
+    .lock()
+    .map_err(|_e| generic_error("cannot acquire poisoned lock"))?
```

Comment:
Is there a better way to handle this error? I tried `e.into()` but could not figure out the compiler error.